solr/core/src/java/org/apache/solr/handler/admin/CollectionsHandler.java

```diff
@@ -419,8 +418,46 @@ private static void addStatusToResponse(NamedList<Object> results, RequestStatus
      */
     CREATE_OP(CREATE, (req, rsp, h) -> {
       req.getParams().required().getAll(null, NAME);
```

Comment:
This can probably go back to  Map<String, Object> props = req.getParams().required().getAll(null, NAME); and remove name from below as it was in the original code. One less map created that way. I had made changes here but I think I incompletely reverted this in the patch I provided.

Reply:
I simply wish to keep it as it was (thus unrefactor). So we agree but I forget the details of how it was.  As I build up the final commit to actually push I'll be mindful to avoid diffs here we don't find useful.  I did make a change below to use putIfAbsent that is worth keeping.